정민지/mvvmstudy/app/src/main/java/com/github/sookhee/mvvmstudy/network/mapper/ResponseMapper.kt

```diff
@@ -0,0 +1,31 @@
+package com.github.sookhee.mvvmstudy.network.mapper
+
+import android.annotation.SuppressLint
+import com.github.sookhee.mvvmstudy.model.GithubRepositoryModel
+import com.github.sookhee.mvvmstudy.network.spec.GithubRepositoryResponse
+import java.text.SimpleDateFormat
+
+/**
+ *  ResponseMapper.kt
+ *
+ *  Created by Minji Jeong on 2022/05/18
+ *  Copyright © 2022 GwanakMT All rights reserved.
+ */
+
+object ResponseMapper {
+    @SuppressLint("SimpleDateFormat")
+    fun mapToGithubRepositoryModelList(response: List<GithubRepositoryResponse>): List<GithubRepositoryModel> {
+        val dateFormat = SimpleDateFormat("yyyy년 MM월 dd일 hh:mm:ss")
+
+        return response.map { repository ->
+            GithubRepositoryModel(
+                id = repository.id,
+                repoName = repository.name,
+                repoLastUpdate = repository.lastUpdate?.let { dateFormat.format(it) } ?: "",
+                language = repository.language ?: "",
+                ownerName = repository.owner.name ?: "",
+                profileImage = repository.owner.profileImage ?: ""
+            )
+        }
+    }
+}
```

Comment:
개인적으로 저는 mapper를 확장함수를 사용해서 정의하고 있어요!
object 변수나 관리할 상태가 없어서 extesions으로 정의하고 있는데 크게 좋은점은 모르겠네요 ㅋㅋㅋㅋ 참고만 해주세요

```suggestion
    @SuppressLint("SimpleDateFormat")
    fun List<GithubRepositoryResponse>.mapToGithubRepositoryModelList(): List<GithubRepositoryModel> {
        val dateFormat = SimpleDateFormat("yyyy년 MM월 dd일 hh:mm:ss")

        return map { repository ->
            GithubRepositoryModel(
                id = repository.id,
                repoName = repository.name,
                repoLastUpdate = repository.lastUpdate?.let { dateFormat.format(it) } ?: "",
                language = repository.language ?: "",
                ownerName = repository.owner.name ?: "",
                profileImage = repository.owner.profileImage ?: ""
            )
        }
    }
```